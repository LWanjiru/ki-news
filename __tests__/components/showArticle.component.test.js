import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import ShowArticle from './../../src/scripts/components/showArticle';
import ShowIframe from './../../src/scripts/components/showIframe';

it('should pass a selected value to the handleClick function', () => {
  const component = shallow(<ShowArticle />);
  component.find('button').simulate('change', { target: {
    value: 'handleClick function' },
  });
  expect(toJson(component)).toMatchSnapshot();
});
it('should render an Iframe', () => {
  const wrapper = shallow(<ShowArticle />);
  expect(wrapper.containsMatchingElement(<ShowIframe />)).toBe(true);
});
it('should change the state', () => {
  const button = shallow(<ShowArticle />);
  expect(button.text()).toEqual('Click Here To View Full Story in an Iframe<ShowIframe />');
  button.find('button').simulate('click');
  expect(button.text()).toEqual('Close Frame');
});

