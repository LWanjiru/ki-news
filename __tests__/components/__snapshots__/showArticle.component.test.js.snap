// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass a selected value to the handleClick function 1`] = `
<div
  className="iframe-btn"
>
  <button
    aria-controls="article-frame"
    aria-expanded="false"
    className="btn btn-success active"
    data-target="#article-frame"
    data-toggle="collapse"
    onClick={[Function]}
    type="button"
  >
    Click Here To View Full Story in an Iframe
  </button>
  <div
    className="collapse"
    id="article-frame"
  >
    <ShowIframe />
  </div>
</div>
`;
